Unit Tests/Tests/Buffer/ConcreteBufferTests.swift

```diff
@@ -0,0 +1,503 @@
+import XCTest
+import TestUtilities
+@testable import PusherChatkit
+
+class ConcreteBufferTests: XCTestCase {
+    
+    // MARK: - Tests
+    
+    func test_init_withIncompleteState_hasNilCurrentState() {
```

Comment:
I will write a dedicated test for that.

Reply:
Actually forget about the above comment. Because we are not passing a delegate in the initializer, it doesn't matter whether `delegate` is being called or not.

Reply:
Exactly.  👍

Reply:
Actually, if you wouldn't mind, can we `XCTAssertNil(sut.delegate)` in all of the `test_init_` tests. 

(This ensures that the class never sets its own delegate in the initialiser)

Reply:
I've added additional assertions.